record/record.go

```diff
@@ -24,11 +24,12 @@
 //		records := record.NewReader(r)
 //		for {
 //			rec, err := records.Next()
-//			if err == io.EOF {
+//			if err == io.EOF || err == io.ErrUnexpectedEOF {
 //				break
 //			}
 //			if err != nil {
-//				return nil, err
+//				r.Recover()  // Skip corrupted records.
+//				continue
 //			}
 //			s, err := ioutil.ReadAll(rec)
 //			if err != nil {
```

Comment:
Updated the code to reflect that. The process is the same. if ioutil.ReadAll gets a non io.EOF or io.ErrUnexpectedEOF, then we can't recover, otherwise we should call r.Recover() and start with a new block/chunk.